analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -332,6 +332,19 @@ static bool TakesExpressionTree(SymbolInfo info)
             return false;
         }
 
+        // based on Type="ArgumentListSyntax" in https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Syntax/Syntax.xml
+        public static ArgumentListSyntax ArgumentList(this SyntaxNode node) =>
```

Comment:
I'm not so sure about it. I intentionally removed it because it seemed a bit odd to me. There was only one rule affected by this change, so I think it is not worth adding support for it.